backend/src/index.js

```diff
@@ -5,140 +5,71 @@
 
 'use strict';
 
-const PORT = process.env.PORT || 3000;
-
 const app = require( 'express' )();
 const { Server } = require( 'http' );
 const http = new Server( app );
 const io = require( 'socket.io' )( http );
+const redisAdapter = require( 'socket.io-redis' );
+const RedisDriver = require( './RedisDriver' );
+const SessionRepository = require( './SessionRepository' );
+const dotenv = require( 'dotenv' );
+dotenv.config( { path: '.env' } );
 
-const db = new Map();
-
-// Priorities are set from the lowest to the highest.
-const statePriorities = [ 'away', 'viewing', 'commenting', 'editing', 'merging' ];
+const config = {
+	host: process.env.REDIS_HOST,
+	port: process.env.REDIS_PORT
+};
 
-io.on( 'connection', socket => {
-	socket.session = {};
+const redisDriver = new RedisDriver( config );
+redisDriver.connect();
 
-	const timestamp = new Date();
+io.adapter( redisAdapter( config ) );
```

Comment:
Should be used only when Redis is available.

Reply:
Done.